### PR TITLE
[Docs] Style guide: General writing and Teleport style suggestions

### PR DESCRIPTION
**Pull request by lsgunn-teleport**
Proposed additions and changes to the _General style rules_ section.

### CODE REVIEW

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 1]
```diff
@@ -150,71 +309,67 @@ Some guides are intended for end users seeking to access resources in their
...
-
-- Write sparser one to two-sentence groupings rather than lengthier blocks of text.
-- Use periods at the end of a line even in a list unless the ending item is a command.
-
 ### Code, commands, and configuration
 
 - `tsh`, `tctl`, and other core commands should be placed in backticks.
 
-- All ports or values should be enclosed in backticks, e.g., `443`.
+- All ports or values should be enclosed in backticks, for example, `443`.
```
Comment (zmb3) [inline docs/pages/contributing/documentation/style-guide.mdx:316]: I'm not a fan of this one. I would prefer we use backticks when we really want a monospaced font (things like code snippets, configuration files, commands, etc). Otherwise we just add extra visual load without good reason.

For example, is:

Teleport listens on port `3080` by default.

really better than

Teleport listens on port 3080 by default.

??

**Review by zmb3 — APPROVED** [review]

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 2]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+- Use an *active voice* and *present tense* whenever possible.
+- Use a *conversational tone* that is not too formal or too chummy.
 
-If a commonly debated style question does not have a resolution in this guide
-(e.g., the Oxford comma), all we ask is that you keep your style consistent
-within a particular page to maintain a professional polish.
+### Point of view
 
-</Notice>
+In most cases, address the reader directly. For tutorials, use either first person
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:122]: Should this be "For tutorials and how-to guides"?

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 3]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
 
-The documentation should be technically precise and directed toward a technical
-audience, e.g., application developers, site reliability engineers, and security
-engineering team leads. 
+Don't use “I” or “me” unless it appears in the text of a user interface element.
 
-Even when describing Teleport generally, we should emphasize specific technical
-capabilities over broad statements of benefit. The aim is not to pique the
-audience's interest, but to provide information.
+Avoid using “we” unless you have a really good rationale for it. For example, if you
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:133]: Does this contradict the advice for tutorials two paragraphs above?

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 4]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
-audience's interest, but to provide information.
+Avoid using “we” unless you have a really good rationale for it. For example, if you 
+are documenting a recommended setting or practice, using “We recommend...” is ambiguous
+("we" being who exactly?) and potentially dangerous if something goes wrong. If there
+are industry best practices or recommendations, it's okay to indicate that instead of 
+using the royal or editorial "we" in documentation.
+
+### Passive constructions
+
+In spite of the axiom to never use the passive voice, there are situations where a
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:141]: Judging by the PRs I review, I'm not sure the reader is necessarily familiar with what passive voice is or that writing guides tend to proscribe it.

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 5]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+
+In spite of the axiom to never use the passive voice, there are situations where a 
+passive sentence structure might be appropriate. Don't twist a sentence into knots just 
+to avoid a passive construction. Passive voice does have its place, but be wary of using it.
+
+When writing about software, it's often tempting to describe what's happening from the 
+code point of view. However, there's almost always a human being with a goal or task to 
+complete who is initiating the activity that the software is executing. If you keep this 
+human presence in mind, your writing will be more dynamic, easier to follow, and more 
+interesting to read.
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:149]: I would put this paragraph first in this section and include examples, clarifying that the default would be to avoid passive voice for prose that's more dynamic, easier to follow, etc., but not so much so that you would tie a sentence in knots. We should include examples because this isn't an idea that all docs authors are familiar with.

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 6]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+
+### Contractions and conversational tone
+
+Contractions are generally acceptable because they give documentation a more natural 
+conversational tone—at least for English speakers. Be conscious of when and why you use 
+contractions.
+
+To keep the tone conversational but concise, adhere to the following common-sense guidelines:
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:157]:
```suggestion
To keep the tone conversational but concise, adhere to the following guidelines:
```
We wouldn't need guidelines if they were common sense.

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 7]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+
+- Use common, well-known words whenever possible.
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:159]: I would include examples here. I'm not sure "common, well-known words" has a self-evident meaning in the context of a specialist tool like Teleport.

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 8]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+
+- Don't use flowery language or literary flourish words and phrases like “and so forth”,
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:161]: I would say these are more academic than literary. We're not avoiding all literary language if we already say, "Don't twist a sentence into knots".

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 9]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+
+- Try to be precise in word choice. For example:
+
+  - Don't use “when”—implying eventuality and time—as interchangeable with “if”, which implies 
+    the possibility of something happening.
+  - Don't use phrases that introduce ambiguity. For example, instead of “When the process completes…” 
+    use “After the process completes…”.
+  - Think carefully about word choices like using “since” (implying a period of time) instead 
+    of “because” (implying cause and result) or using “once” (single occurrence) instead of 
+    “after” (each time).
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:172]: Does this list belong in a style guide? This sounds like general advice for writing, and I'm not sure that professional writing even consistently avoids the bad examples. I think our style guide should be as short as it can reasonably be to make it easier for authors to follow.

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 10]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+
+- Avoid using dead language words and phrases even if they are generally accepted as English 
+  words in practice. For example:
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:175]: What's the reasoning for this? English is filled with load words, so this strikes me as somewhat arbitrary and limiting.

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 11]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+
+- Avoid adding unnecessary words or phrases. For example:
+
+  - Instead of “In order to”, just use “to”.
+  - Instead of “as well as”, just use “and”.
+  - Instead of “and then”, just use “then”.
+  - Avoid jargon, colloquialisms, and idiomatic phrases.
+
+- Avoid adverbs and subjective statements. For example, don't use words and phrases like 
+  easily, rapidly, simply, quickly.
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:195]:
```suggestion
  "easily," "rapidly," "simply," and "quickly."
```

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 12]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+### Headings
+
+All heading levels should use the following conventions:
+
+- Use sentence case. Sentence case means that only the initial word and proper nouns are 
+  capitalized.
+- Use active, present tense verbs in headings wherever appropriate, especially in the 
+  context of tutorials and how-to guides.
+- Use noun phrases for conceptual and reference topics whenever possible.
+- Summarize as succinctly as possible the content they contain.
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:209]:
```suggestion
- Summarize as succinctly as possible the section they introduce.
```
This threw me a little. I was wondering, "How can a heading summarize itself?"

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 13]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+- Avoid generic headings like overview and introduction, if possible. While generic
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:210]:
```suggestion
- Avoid generic headings like "Overview" and "Introduction," if possible. While generic 
```

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 14]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+
+### Limit heading levels
+
+As a best practice, avoid building an information hierarchy with more than three heading 
+levels. Most content can be effectively organized using two internal heading levels, making 
+it easier to navigate and scan for relevant topics. 
+
+### Topic titles
+
+Avoid using gerunds (verbs ending in -ing) in titles and headings. Procedure titles and
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:224]:
```suggestion
Avoid using gerunds (verbs ending in "-ing") in titles and headings. Procedure titles and 
```

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 15]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+### Limit heading levels
+
+As a best practice, avoid building an information hierarchy with more than three heading 
+levels. Most content can be effectively organized using two internal heading levels, making 
+it easier to navigate and scan for relevant topics. 
+
+### Topic titles
+
+Avoid using gerunds (verbs ending in -ing) in titles and headings. Procedure titles and 
+headings should answer the question: *What are you trying to do?* For example, if the answer
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:225]:
```suggestion
headings should answer the question: *What are you trying to do?*. 

For example, if the answer 
```

File: docs/pages/contributing/documentation/style-guide.mdx [hunk 16]
```diff
@@ -109,30 +109,189 @@ Specific kinds of architecture guides include:
...
+  - Think carefully about word choices like using “since” (implying a period of time) instead 
+    of “because” (implying cause and result) or using “once” (single occurrence) instead of 
+    “after” (each time).
+
+- Avoid using dead language words and phrases even if they are generally accepted as English 
+  words in practice. For example:
+
+  - Instead of “i.e.”, use “that is” or rewrite the sentence to make the meaning clear without 
+    needing extra qualification.
+  - Instead of “e.g.”, use “for example”.
```
Comment (ptgott) [inline docs/pages/contributing/documentation/style-guide.mdx:179]: In parenthetical statements consisting only of examples, I think "e.g.," makes more sense and appears more succinct than "for example."

File: docs/pages/contributing/documentation/style-guide.mdx (hunk 10 above)
Comment (lsgunn-teleport) [reply docs/pages/contributing/documentation/style-guide.mdx:175]: For clarity, expressing what the dead language or abbreviation means in plain English ensures that everyone understands the intended meaning.
For translation (not an issue for Teleport doc perhaps), using plain English can help translators understand the intended meaning and pick an appropriate term.
